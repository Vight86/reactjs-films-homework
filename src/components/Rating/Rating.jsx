import React from 'react';
import propTypes from 'prop-types';
import styles from './Rating.scss';

const Rating = ({
  className, withStars, children,
}) => {
  const width = (26 / 5) * Number(parseFloat(children).toFixed());
  const ratingNumber = Number(((children / 10) * 5).toFixed(1));
  let rating;

  if (withStars === true) {
    rating = (
      <div className={styles.rating}>
        <span className={styles.stars} style={{ width: `${width}em` }} />
        <span className={styles[className]}>
          {ratingNumber}
        </span>
      </div>
    );
  } else {
    rating = (
      <span className={styles[className]}>
        {ratingNumber}
      </span>
    );
  }

  return rating;
};

Rating.propTypes = {
  className: propTypes.string,
  withStars: propTypes.bool,
  children: propTypes.number,
};

Rating.defaultProps = {
  className: '',
  withStars: false,
  children: 0,
};

export default Rating;
