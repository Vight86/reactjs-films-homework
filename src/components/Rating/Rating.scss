@import '../../utils/variables';

.rating {
  width: 2.5em;
  height: 1.5em;
  margin: 0 0 0 auto;
  border-radius: 0.17em;
  font-size: 2.4em;
  font-weight: 500;
  line-height: 1.5em;
  text-align: center;
}

.primary {
  @extend .rating;

  border: 2px solid $color-tertiary;
  margin: 0;
  color: $color-light;
}

.secondary {
  @extend .rating;

  border: 2px solid currentColor;
  color: $color-tertiary;
}
