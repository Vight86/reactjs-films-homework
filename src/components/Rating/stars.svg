<svg xmlns="http://www.w3.org/2000/svg" width="236" height="36" viewBox="0 0 236 36">
  <defs>
    <style>
      .cls-1 {
        fill: #0aaee4;
        fill-rule: evenodd;
      }
    </style>
  </defs>
  <path id="Shape_1_copy_4" data-name="Shape 1 copy 4" class="cls-1" d="M36,13.5H22.22L18,0,13.783,13.5H0l11.165,8.374L6.75,36,18,27l11.25,9L24.836,21.873Zm50,0H72.22L68,0l-4.22,13.5H50l11.165,8.374L56.75,36,68,27l11.25,9L74.836,21.873Zm50,0H122.22L118,0l-4.219,13.5H100l11.165,8.374L106.75,36,118,27l11.25,9-4.414-14.127Zm50,0H172.22L168,0l-4.219,13.5H150l11.165,8.374L156.75,36,168,27l11.25,9-4.414-14.127Zm50,0H222.22L218,0l-4.219,13.5H200l11.165,8.374L206.75,36,218,27l11.25,9-4.414-14.127Z"/>
</svg>
