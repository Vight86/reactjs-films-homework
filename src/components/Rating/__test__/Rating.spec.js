import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Rating from '../index';

const setUp = (className, children) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Rating className={className}>{children}</Rating>);
  return renderer.getRenderOutput();
};

describe('render Rating component', () => {
  it('component tag should be p', () => {
    const result = setUp();
    expect(result.type).toBe('p');
  });

  it('render Rating className correctly', () => {
    const result = setUp('test className', null);
    expect(result.props.className).toBe('test className');
  });

  it('render Rating children correctly', () => {
    const result = setUp(null, 'test children');
    expect(result.props.children).toBe('test children');
  });
});
