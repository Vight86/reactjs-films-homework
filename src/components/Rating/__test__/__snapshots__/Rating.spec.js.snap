// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Rating component render with stars correctly 1`] = `
<div
  className="rating"
>
  <span
    className="stars"
    style={
      Object {
        "width": "20.8em",
      }
    }
  />
  <span
    className="test"
  >
    2
  </span>
</div>
`;

exports[`render Rating component render without stars correctly 1`] = `
<span
  className="test"
>
  2
</span>
`;
